addons/checkparam/checkparam.lua

```diff
@@ -0,0 +1,305 @@
+  THF = 'store tp|double attack|triple attack|quadruple attack|dual wield',
+  PLD = 'enmity|damage taken|physical damage taken|magic damage taken|spell inturruption rate|pharanx|cure potency|fastcast',
+  DRK = 'store tp|double attack|triple attack|quadruple attack|weapon skill damage',
+  BST = 'pet: double attack|pet: magic attack bonus|pet: damage taken',
+  BRD = 'all songs|song effect duration|fast cast|song spellcasting time',
+  RNG = 'store tp|snapshot|rapid shot|weapon skill damage',
+  SAM = 'store tp|double attack|triple attack|quadruple attack|weapon skill damage',
+  NIN = 'store tp|double attack|triple attack|quadruple attack|subtle blow',
+  DRG = 'store tp|double attack|triple attack|quadruple attack|weapon skill damage',
+  SMN = 'blood pact delay|blood pact delay ii|blood pact damage|avatar perpetuation cost|pet: magic attack bonus|pet: str|pet: double attack',
+  BLU = 'store tp|double attack|triple attack|quadruple attack|critical hit rate|critical hit damage|weapon skill damage|fast cast|magic attack bonus|magic accuracy|cure potency',
+  COR = 'store tp|snapshot|rapid shot|fast cast|cure potency|magic accuracy|magic attack bonus|magic damage|weapon skill damage',
+  PUP = 'pet: hp|pet: damage taken|pet: regen|martial arts|store tp|double attack|triple attack|quadruple attack',
+  DNC = 'store tp|double attack|triple attack|quadruple attack',
+  SCH = 'magic attack bonus|magic burst damage|magic burst damage ii|magic accuracy|magic damage|fast cast|elemental magic casting time|cure potency|enh mag eff dur|enhancing magic effect duration',
+  GEO = 'pet: regen|pet: damage taken|indicolure effect duration|indi eff dur|fast cast|magic evasion',
+  RUN = 'enmity|damage taken|physical damage taken|magic damage taken|spell inturruption rate|pharanx|inquartata|fastcast',
+  levelfilter = 99,
+  debugmode = false,
+}
+settings = config.load(defaults)
+
+tbl = {}
+
+windower.register_event('addon command',function(arg)
+  if arg == 'debugmode' then
+    settings.debugmode = not settings.debugmode
+    settings:save('all')
+  else
+    for i=0,#res.slots do
+      local slot = windower.regex.replace(string.lower(res.slots[i].en),' ','_')
+      local gear_set = windower.ffxi.get_items().equipment
+      local gear = windower.ffxi.get_items(gear_set[slot..'_bag'],gear_set[slot])
+      if gear_set[slot] > 0 then
+        get_text(gear.id,gear.extdata)
+      end
+    end
+    local my = windower.ffxi.get_player()
+    show_results(my.name,my.main_job,my.sub_job)
+  end
+end)
+
+windower.register_event('incoming chunk',function(id,data)
+  if id == 0x0C9 then
+    local p = packets.parse('incoming',data)
+    if p['Type'] == 3 then
+      local count = p['Count']
+      if count == 1 then
+        get_text(p['Item'],p['ExtData'])
+      else
+        for i=1,count do
+          get_text(p['Item '..i],p['ExtData '..i])
+        end
+      end
+    elseif p['Type'] == 1 then
+      local t = windower.ffxi.get_mob_by_id(p['Target ID'])
```

Comment:
`windower.ffxi.get_mob_by_index(p['Target Index'])` is faster than `get_mob_by_id`.